manifests/canonical.pp

```diff
@@ -35,7 +35,7 @@
 #
 define postfix::canonical (
   $destination,
-  $file='/etc/postfix/canonical',
+  $file="${postfix::confdir}/canonical",
```

Comment:
I'm not sure about depending on the `postfix` class in the parameters, as this means users need to instantiate it before they call the definition.

One way to avoid this would be to default to `undef` and then build an internal variable based on `postfix::config` (after including `postfix` to be sure the variable is instantiated).

Reply:
Hum, yes, I consider doing this an anti-pattern but maybe it's safer to do as you say.  I added a commit to do this.  Thanks!

Reply:
There are a bunch of use of other variables before including postfix (and sometimes without inclusion of postfix at all).  I tried to fix all of them in my last commit :wink:.